### ISSUE TITLE
oc-env command errors on fish shell

### ISSUE THREAD

**Issue by quagliero** [minishift/minishift#1006]
```fish
➜ ~ eval (minishift oc-env)
Unexpected end of string, quotes are not balanced
- (line 1): begin; set -gx PATH $PATH /home/q/.minishift/cache/oc/v1.5.1"; # Run this command to configure your shell:  # eval (minishift oc-env) 
                                                                             ^
from sourcing file -
	called on line 60 of file /usr/share/fish/functions/eval.fish

in function “eval”
	called on standard input

source: Error while reading file “-”

```
This appears to be due to an unnecessary closing quote on https://github.com/minishift/minishift/blob/master/pkg/util/shell/shell.go#L77. If that is indeed the solution
I shall submit a PR shortly.

**Comment by hferentschik** [comment]
@quagliero thanks for the issue report and pull request. Much appreciated. The quotes are indeed unbalanced. My question though is why you think that the closing quote should be removed over adding a actual opening quote. I am not a fish user, so hence I am asking. If this were bash I clearly say we should be adding a opening quote.

**Comment by quagliero** [comment]
@hferentschik as far as I'm aware both would work, but the fish docs themselves don't use quotes (https://fishshell.com/docs/current/tutorial.html#tut_exports), and I never have when adding things to the $PATH in fish so didn't consider adding one :smile:

**Comment by hferentschik** [comment]
What's about multiple values though and potential spaces in them? The code we are looking at just recently got refactored and I think a regression got introduced. Before it properly used quotes which is what Minikube uses as well as docker-machine. For example:

    $ docker-machine env --shell=fish xhyve
    ...
    set -gx DOCKER_HOST "tcp://192.168.64.18:2376";
    ...

Also looking at the current output for fish, I think there is another issue:

    minishift oc-env --shell=fish
    set -gx PATH $PATH /Users/foo/.minishift/cache/oc/v1.5.1";

We actually should add the oc path to the from of the PATH (as is done in the other shell types). If I read this properly, for fish it would be added at the end. 

@budhrg, wdyt? Could you help sorting this out?

**Comment by hferentschik** [comment]
See also https://github.com/docker/machine/blob/master/commands/env.go

**Comment by budhrg** [comment]
>  Could you help sorting this out?

@hferentschik , will sort this today, verify the PR and merge it.

**Comment by budhrg** [comment]
> We actually should add the oc path to the from of the PATH (as is done in the other shell types). If I read this properly, for fish it would be added at the end.

Yes, we can do that `set -gx PATH /Users/foo/.minishift/cache/oc/v1.5.1 $PATH;

**Comment by quagliero** [comment]
Updated and rebased.

**Comment by hferentschik** [comment]
> Yes, we can do that set -gx PATH /Users/foo/.minishift/cache/oc/v1.5.1 $PATH;

I did not say, that you cannot do it. In fact in most cases it will work just fine. However, this we are interpolating a variable ($PATH) we might have spaces in it (unlikely but who knows). Using " is more defensive. 

If nothing else there is an argument for aligning with Minikube and docker-machine. 

@quagliero, if you like feel free to update your pull request to add quotes. If not, we can merge your pull request and create a follow up issues. I'd like to also add several unit tests which will ensure that we are not breaking the intended format anymore.

**Comment by quagliero** [comment]
@hferentschik update pushed.

**Comment by hferentschik** [comment]
Resolved via pull request #1007
